node/tests/SharedTests.ts

```diff
@@ -578,6 +578,35 @@ export function runBaseTests<Context>(config: {
         config.timeout,
     );
 
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `testing hkeys with exiting, an non exising key and error request key_%p`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key = uuidv4();
+                const key2 = uuidv4();
+                const field1 = uuidv4();
+                const field2 = uuidv4();
+                const value = uuidv4();
+                const value2 = uuidv4();
+                const fieldValueMap = {
+                    [field1]: value,
+                    [field2]: value2,
+                };
+
```

Comment:
comment each scenario why we expect the result
e.g. if key is not hash so we expect error